src/test/java/com/iota/iri/network/impl/TransactionRequesterWorkerImplTest.java

```diff
@@ -0,0 +1,142 @@
+import com.iota.iri.controllers.TransactionViewModelTest;
+import com.iota.iri.model.Hash;
+import com.iota.iri.model.persistables.Transaction;
+import com.iota.iri.network.Node;
+import com.iota.iri.network.TransactionRequester;
+import com.iota.iri.network.impl.TransactionRequesterWorkerImpl;
+import com.iota.iri.storage.Tangle;
+import com.iota.iri.utils.Converter;
+
+public class TransactionRequesterWorkerImplTest {
+    
+    //Good
+    private static final TransactionViewModel TVMRandomNull = new TransactionViewModel(
+            getRandomTransaction(new Random()), Hash.NULL_HASH);
+    private static final TransactionViewModel TVMRandomNotNull = new TransactionViewModel(
+            getRandomTransaction(new Random()), TransactionViewModelTest.getRandomTransactionHash());
+    private static final TransactionViewModel TVMAll9Null = new TransactionViewModel(
+            get9Transaction(), Hash.NULL_HASH);
+    
+    //Bad
+    private static final TransactionViewModel TVMAll9NotNull = new TransactionViewModel(
+            get9Transaction(), TransactionViewModelTest.getRandomTransactionHash()); 
+    
+    private static Tangle tangle;
+    private static Node node;
+    private static TipsViewModel tvm;
+    
+    private static TransactionRequester requester;
+    private static TransactionRequesterWorkerImpl worker;
+    
+    @Rule 
+    public MockitoRule mockitoRule = MockitoJUnit.rule();
+    
+    @BeforeClass
+    public static void setUp(){
+        requester = mock(TransactionRequester.class);
+        requester.init(0); //Dont drop anything
+    }
+
+    @Before
+    public void before() throws NoSuchFieldException, SecurityException, Exception {
+        tangle = mock(Tangle.class);
+        tvm = mock(TipsViewModel.class);
+        
+        worker = new TransactionRequesterWorkerImpl();
+        
+        node = mock(Node.class);
+        when(node.getNeighbors()).thenReturn(new LinkedList<>());
+        
+        //Set start queue to 0 for mocking
+        setFinalStatic(worker.getClass().getDeclaredField("REQUESTER_THREAD_ACTIVATION_THRESHOLD"), 0);
+        worker.init(tangle, requester, tvm, node);
+    }
+    
+    @AfterClass
+    public static void tearDown() throws Exception {
+        worker.shutdown();
+    }
+    
+    @Test
+    public void processRequestQueueTest() throws Exception {
+        TransactionRequesterWorkerImpl fakeWorker = mock(TransactionRequesterWorkerImpl.class);
+        
+        when(fakeWorker.isActive()).thenReturn(true);
+        when(fakeWorker.getTransactionToSendWithRequest())
+            .thenReturn(TVMRandomNull)
+            .thenReturn(TVMRandomNotNull)
+            .thenReturn(TVMAll9Null)
+            .thenReturn(TVMAll9NotNull);
+        
+        when(fakeWorker.init(any(), any(), any(), any())).thenCallRealMethod();
+        fakeWorker.init(tangle, requester, tvm, node);
+        
+        when(fakeWorker.isValidTransaction(any())).thenCallRealMethod();
```

Comment:
hmm once you use the real impl you can return the scope of the method back to private I think :-)